protos/local/failure.proto

```diff
@@ -0,0 +1,77 @@
+syntax = "proto3";
+
+package coresdk.failure;
+
+import "google/protobuf/duration.proto";
+import "common.proto";
+
+/// Represents a failure in user code, workflow or activity, which could've been triggered by
+/// an exception or similar error mechanism like the error half of a Result type.
+message ApplicationFailureInfo {
```

Comment:
Can we keep calling this `UserCodeFailure` ? I think that name is way more clear.

Also we can probably drop `Info` off of all these names. Not useful.

Reply:
Let's keep the `ApplicationFailure` name for compatibility with the other SDKs.